src/sys/socket/mod.rs

```diff
@@ -2441,4 +2449,15 @@ mod tests {
         )
         .expect("Failed to open routing socket");
     }
+    #[test]
+    fn sock_protocol_constant_matches() {
```

Comment:
This part is tautological.  I suggest removing it.

Reply:
I wouldn't say it is tautological.
We're initializing `CanRaw` to `Icmp`, because these constants happen to match; this is an added sanity to make sure this is correct. (I'd have opted for a `static_assert!` if it was available).

I do think it is unlikely to ever break, so will remove it.